Geometry/MuonNumbering/src/DD4hep_MuonNumbering.cc

```diff
@@ -18,7 +18,11 @@ const MuonBaseNumber MuonNumbering::geoHistoryToBaseNumber(const cms::ExpandedNo
     edm::LogError("Geometry") << "MuonNumbering finds unusual base constant:" << basePart;
   }
   if (superPart < 100) {
-    edm::LogError("Geometry") << "MuonNumbering finds unusual super constant:" << superPart;
+    edm::LogError("Geometry") << "MuonNumbering finds unusual super part:" << superPart;
+    if (superPart <= 0) {
+      edm::LogError("Geometry") << "MuonNumbering -- resetting super part to 100";
+      superPart = 100;  // Protect against division by 0
```

Comment:
I imagine that this is just hypothetical, and only made in order to quiet down the static analyzer: but why "100", and not (e.g.) "1"? If superPart is 1 or 2 it remains as such, if it is 0 it goes to 100...

Reply:
I would agree with Andrea - the check (<0) is problematic. May be a correct code would be without this "if" - all superPart should be above 100 or 100.

Reply:
Yes, I'll drop the `<=0` check and combine the two messages.